wgpu-core/src/device/descriptor.rs

```diff
@@ -0,0 +1,168 @@
+    for DescriptorDevice<'_, B>
+{
+    unsafe fn create_descriptor_pool(
+        &self,
+        descriptor_count: &DescriptorTotalCount,
+        max_sets: u32,
+        flags: gpu_descriptor::DescriptorPoolCreateFlags,
+    ) -> Result<B::DescriptorPool, gpu_descriptor::CreatePoolError> {
+        let mut ranges = ArrayVec::<[_; 7]>::new();
+        ranges.push(hal::pso::DescriptorRangeDesc {
+            ty: hal::pso::DescriptorType::Sampler,
+            count: descriptor_count.sampler as _,
+        });
+        ranges.push(hal::pso::DescriptorRangeDesc {
+            ty: hal::pso::DescriptorType::Image {
+                ty: hal::pso::ImageDescriptorType::Sampled {
+                    with_sampler: false,
+                },
+            },
+            count: descriptor_count.sampled_image as _,
+        });
+        ranges.push(hal::pso::DescriptorRangeDesc {
+            ty: hal::pso::DescriptorType::Image {
+                ty: hal::pso::ImageDescriptorType::Storage { read_only: false },
+            },
+            count: descriptor_count.storage_image as _,
+        });
+        ranges.push(hal::pso::DescriptorRangeDesc {
+            ty: hal::pso::DescriptorType::Buffer {
+                ty: hal::pso::BufferDescriptorType::Uniform,
+                format: hal::pso::BufferDescriptorFormat::Structured {
+                    dynamic_offset: false,
+                },
+            },
+            count: descriptor_count.uniform_buffer as _,
+        });
+        ranges.push(hal::pso::DescriptorRangeDesc {
+            ty: hal::pso::DescriptorType::Buffer {
+                ty: hal::pso::BufferDescriptorType::Storage { read_only: false },
+                format: hal::pso::BufferDescriptorFormat::Structured {
+                    dynamic_offset: false,
+                },
+            },
+            count: descriptor_count.storage_buffer as _,
+        });
+        ranges.push(hal::pso::DescriptorRangeDesc {
+            ty: hal::pso::DescriptorType::Buffer {
+                ty: hal::pso::BufferDescriptorType::Uniform,
+                format: hal::pso::BufferDescriptorFormat::Structured {
+                    dynamic_offset: true,
+                },
+            },
+            count: descriptor_count.uniform_buffer_dynamic as _,
+        });
+        ranges.push(hal::pso::DescriptorRangeDesc {
+            ty: hal::pso::DescriptorType::Buffer {
+                ty: hal::pso::BufferDescriptorType::Storage { read_only: false },
+                format: hal::pso::BufferDescriptorFormat::Structured {
+                    dynamic_offset: true,
+                },
+            },
+            count: descriptor_count.storage_buffer_dynamic as _,
+        });
+        ranges.retain(|rd| rd.count != 0);
+
+        match hal::device::Device::create_descriptor_pool(
+            self.0,
+            max_sets as usize,
+            ranges,
+            hal::pso::DescriptorPoolCreateFlags::from_bits_truncate(flags.bits() as u32),
```

Comment:
Even if bits probably will be the same as both gfx and gpu-descriptor tend to use same bits as Vulkan,
it is not guaranteed.

Reply:
Can we guarantee that? I mean, at least specify in documentation of gpu-alloc and gpu-descriptor that the bits match Vulkan.

Reply:
Done